src/platform/plugins/shared/unified_search/public/query_string_input/query_bar_top_row.tsx

```diff
@@ -284,6 +284,7 @@ export const QueryBarTopRow = React.memo(
     const isMobile = useIsWithinBreakpoints(['xs', 's']);
     const [isXXLarge, setIsXXLarge] = useState<boolean>(false);
```

Comment:
[non-blocking suggestion] since `isXXLarge` is only used to define `submitButtonIconOnly`, and since `isXXLarge` is checking against the breakpoint `m` now, I think you can replace it with `!isMobile` and remove the entire `useEffect`…?

(this does not affect the output, only the implementation)